determined-ai/determined
# feat: open tasks in existing tab [WEB-420]

## Description

feat: open tasks in existing tab [WEB-420]

Tasks will now either open in a new tab or open in an existing tab for the task, instead of the current behavior of always opening a new tab. 




## Test Plan

1. Launch a JupyterLab instance. A notebook should open in a new page. Keep the tab open.
2. In another tab, on the tasks page click on the new tasks "Name". The browser should open the currently existing tasks tabs, not a new tab. 
3. In another tab, on the tasks page click on the new tasks "Short ID". The browser should open the currently existing tasks tabs, not a new tab. 
4. Open a Tensorboard instance for an experiment. Tensorboard should open in a new page. Keep the tab open.
5. In another tab, on the tasks page click on the new tasks "Name". The browser should open the currently existing tasks tabs, not a new tab. 
6. In another tab, on the tasks page click on the new tasks "Short ID". The browser should open the currently existing tasks tabs, not a new tab. 




## Commentary (optional)

To implement this feature the `noopener noreferrer` window features were removed from task related links. This change is sensible since tasks now open inside of the WebUI instead of redirecting to another domain. 




## Checklist
~- [ ] Changes have been manually QA'd~
~- [ ] User-facing API changes need the "User-facing API Change" label.~
~- [ ] Release notes should be added as a separate file under `docs/release-notes/`.~
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.~
~- [ ] Licenses should be included for new code which was copied and/or modified from any external code.~





[WEB-420]: https://determinedai.atlassian.net/browse/WEB-420?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
